Tests/Marketplace/marketplace_services.py

```diff
@@ -4324,3 +4328,53 @@ def is_content_item_in_id_set(display_name: str, rn_header: str, id_set: dict, m
 
     logging.debug(f"Could not find the entity with display name {display_name} in id_set.")
     return False
+
+
+def remove_old_versions_from_changelog(changelog: dict):
+    """
+    Removes old pack versions from changelog in order to reduce index.zip size.
+    We are keeping the maximum number of versions between the following options:
+    1.  Versions were released last year.
+    2.  Last minor version and one version before it.
+    3.  Last five versions.
+    Edits the changelog entries in place.
+
+    Args:
+        changelog (dict): The changelog of some pack.
+    Returns:
+        (list) last pack versions
+    """
+    versions_to_keep: List[str] = []
+    if not changelog:
+        return versions_to_keep
+
+    last_same_minor_versions = []
+    last_year_versions = []
+    last_five_versions = list(changelog.keys())[-5:]
+
+    year_ago_datetime_obj = datetime.utcnow() - timedelta(days=365)
+    last_version = Version(list(changelog.keys())[-1])
+    save_last_minor_versions = not (last_version.minor == 0 and last_version.major == 1)
+
+    prev_version = None
+    for version, info in changelog.items():
```

Comment:
let's make sure the iteration is sorted by `key=lambda v: Version(v)`

Reply:
The iteration is sorted by the order it is in the changelog. When reading from json, the order is saved.